### PR TITLE
Adding OCK demo getting_started.md

### PR DESCRIPTION
**Pull request by MaryaSharf**
# Overview

Adding a getting started guide for the new OCK_demo

# Description of change

Documentation is added for the OCK demo. Also some changes in the OCK_demo workflow
have been made. Testing for oneAPI samples have been added and artifacts have been updated.

### CODE REVIEW

File: examples/technical_blogs/ock_demo_blog/getting_started.md [hunk 1]
```diff
@@ -0,0 +1,528 @@
+Getting Started
+===
+
+## Introduction
+
+The release package is a self-contained pre-built project to demonstrate how to run OpenCL
+ and SYCL applications on the RISC-V spike simulator, including a deep neural
```
Comment (coldav) [inline examples/technical_blogs/ock_demo_blog/getting_started.md:7]: weird indentation here

File: examples/technical_blogs/ock_demo_blog/getting_started.md [hunk 2]
```diff
@@ -0,0 +1,528 @@
...
+of OpenCL and SYCL applications.
+
+The release package contains a number of different directories:
+
+- oneAPI Construction Kit(OCK) is Codeplay’s multi-target, multi-platform framework.
+    - A few examples have been added to OCK that are useful for analyzing and demonstrating
+      the performance of OCK and oneAPI on RISC-V.
+- portDNN and portBLAS network sample project that executes VGG and Resnet50 networks
+
+```Note: cuda-to-sycl-nbody is in the process of being included to the new OCK demo package```
```
Comment (coldav) [inline examples/technical_blogs/ock_demo_blog/getting_started.md:20]: scans weird, and I'm not sure you need to say this.

File: examples/technical_blogs/ock_demo_blog/getting_started.md [hunk 3]
```diff
@@ -0,0 +1,528 @@
...
+## Installation and Setup
+
+OCK demo package contains the build and install directories of all components. All the repositories mentioned above are open-sourced and can be cloned as follows.
+
+- [oneAPI Construction Kit](https://github.com/codeplaysoftware/oneapi-construction-kit.git)
+- [portDNN](https://github.com/codeplaysoftware/portDNN.git)
+- [portBLAS](https://github.com/codeplaysoftware/portBLAS.git)
+- [oneAPI Samples](https://github.com/oneapi-src/oneAPI-samples)
+- [oneAPI toolkit](https://github.com/intel/llvm/releases)
+- [cuda-to-sycl-nbody](https://github.com/codeplaysoftware/cuda-to-sycl-nbody.git)
```
Comment (coldav) [inline examples/technical_blogs/ock_demo_blog/getting_started.md:40]: not there (although maybe above is okay)

File: examples/technical_blogs/ock_demo_blog/getting_started.md [hunk 4]
```diff
@@ -0,0 +1,528 @@
...
+written using portDNN, SYCL and C++.
+
+## Installation and Setup
+
+OCK demo package contains the build and install directories of all components. All the repositories mentioned above are open-sourced and can be cloned as follows.
+
+- [oneAPI Construction Kit](https://github.com/codeplaysoftware/oneapi-construction-kit.git)
+- [portDNN](https://github.com/codeplaysoftware/portDNN.git)
+- [portBLAS](https://github.com/codeplaysoftware/portBLAS.git)
+- [oneAPI Samples](https://github.com/oneapi-src/oneAPI-samples)
```
Comment (coldav) [inline examples/technical_blogs/ock_demo_blog/getting_started.md:38]: I don't think we need oneAPI Samples it's too big, and we now have our own examples

File: examples/technical_blogs/ock_demo_blog/getting_started.md [hunk 5]
```diff
@@ -0,0 +1,528 @@
...
+The above repositories can be built for dpc++ following their respective build instructions. 
+All of the following instructions have been tested on Ubuntu:20.04.
+
+You'll need to extract the files from the archive file:
+
+```sh
+    mkdir Release
+    cd Release
+    tar -xf ../ock_demo_artifacts.tar.gz
+    tar -xf ../network_artifacts.tar.gz
```
Comment (coldav) [inline examples/technical_blogs/ock_demo_blog/getting_started.md:51]: I'm still keen on separating out the network code from the data, but that can come later if needed.

File: examples/technical_blogs/ock_demo_blog/getting_started.md [hunk 6]
```diff
@@ -0,0 +1,528 @@
...
+    export RELEASE_DIR=$PWD
+```
+
+Once you have extracted the files from the archive file, you should have a
+hierarchy that looks like this:
+
+```
+- WORKDIR: $RELEASE_DIR
+    - DIR: build_riscv_test/
```
Comment (coldav) [inline examples/technical_blogs/ock_demo_blog/getting_started.md:60]: I'm not sure about some of these names - `build` means little to a user  and install does not mean anything.

File: examples/technical_blogs/ock_demo_blog/getting_started.md [hunk 7]
```diff
@@ -0,0 +1,528 @@
...
+
+This section will be split in four parts:
+- Running the vector-add from oneAPI-samples.
+- Running the VGG16 and ResNet50 neural networks.
+- Running the samples from OCK.
+
+### oneAPI-Samples
+
+oneAPI-samples is a comprehensive repository containing a lot of non-sycl examples, but for the sake of our example, we will be only focusing on the ```vector-add``` one. This sample does what it says, it implements adding two vectors together using SYCL.
+
```
Comment (coldav) [inline examples/technical_blogs/ock_demo_blog/getting_started.md:84]: Is there not a near identical test in our samples?

File: examples/technical_blogs/ock_demo_blog/getting_started.md [hunk 8]
```diff
@@ -0,0 +1,528 @@
...
+There are several ways that you can enable the profiler and store those
+information.
+
+Here's the variables that will be used:
+- ```CA_PROFILE_LEVEL```
+- ```CA_PROFILE_CSV_PATH```
+
+#### CA_PROFILE_LEVEL
+
+The profiler has four distinct levels to perform tracking. Here's how to set
```
Comment (coldav) [inline examples/technical_blogs/ock_demo_blog/getting_started.md:433]: we've kind of mentioned CA_PROFILE_LEVEL above, should we drop it from there or reorder - maybe do the networks last?

File: examples/technical_blogs/ock_demo_blog/getting_started.md (hunk 2 above)
Comment (MaryaSharf) [reply examples/technical_blogs/ock_demo_blog/getting_started.md]: Yes, I'll add it once/if sycl_nbody is added to the ock_demo

File: .github/workflows/create_publish_artifacts.yml [hunk 9]
```diff
@@ -52,17 +52,31 @@ jobs:
...
-      - name: Setup environment and test Vector Add
+      - name: Test RSICV examples
         run: |
-          export ONEAPI_ROOT=$(pwd)/linux_nightly_release
+          # Set environment variables
+          export LD_LIBRARY_PATH=$(pwd)/install/lib:$(pwd)/linux_nightly_release/lib/libsycl.so:$(pwd)/linux_nightly_release/lib:$LD_LIBRARY_PATH
+          export CMAKE_CXX_COMPILER=$(pwd)/linux_nightly_release/bin/clang++
+          export CMAKE_C_COMPILER=$(pwd)/linux_nightly_release/bin/clang
+          export CA_HAL_DEBUG=1
+          #export CA_PROFILE_LEVEL=3
```
Comment (coldav) [inline .github/workflows/create_publish_artifacts.yml:62]: why is this commented out?

Comment (MaryaSharf) [reply .github/workflows/create_publish_artifacts.yml:62]: for testing, uncommenting

File: .github/workflows/create_publish_artifacts.yml [hunk 10]
```diff
@@ -52,17 +52,31 @@ jobs:
...
+          export ONEAPI_DEVICE_SELECTOR=opencl:acc
```
[the remark this answers is not in the document]
Comment (MaryaSharf) [reply .github/workflows/create_publish_artifacts.yml]: tested and updated

**Pull request merged**